### PR TITLE
Update scala intellij tutorial

### PR DESCRIPTION
**Pull request by zachgk**
## Description ##
Fix issues described in #12523 

### Essentials ###
Please feel free to remove inapplicable items for your PR.
- [x] Changes are complete (i.e. I finished coding on this PR)
- [x] To the my best knowledge, examples are either not affected by this change, or have been fixed to be compatible with this change

### Changes ###
- [x] Update mxnet version
- [x] Added dependencies to pom.xml
- [x] Log4j configuration file
- [x] Instructions for integrating a build from source

## Comments ##
- If this change is a backward incompatible change, why must this change be made.
- Interesting edge cases to note here

@lanking520 @andrewfayres @aaronmarkham

### CODE REVIEW

File: docs/tutorials/scala/mxnet_scala_on_intellij.md [hunk 1]
```diff
@@ -73,7 +73,7 @@ The configuration you should update is in the pom file's dependency for MXNet:
 <dependency>
   <groupId>org.apache.mxnet</groupId>
   <artifactId>mxnet-full_2.11-osx-x86_64-cpu</artifactId>
-  <version>1.2.0</version>
+  <version>1.3.0</version>
```
Comment (aaronmarkham) [inline docs/tutorials/scala/mxnet_scala_on_intellij.md]: Can we delete this version line? Won't it pull the latest?

Comment (aaronmarkham) [reply docs/tutorials/scala/mxnet_scala_on_intellij.md]: Note that we don't use in this on the install page for scala.

File: docs/tutorials/scala/mxnet_scala_on_intellij.md [hunk 2]
```diff
@@ -292,11 +311,24 @@ The project's `pom.xml` will be open for editing.
 
 Click "Import Changes" in this prompt.
 
-**Step 5.** Build the project:
+**Step 5.** Setup log4j configuration
+
+Create a new file `target/classes/log4j.properties` with the contents:
```
Comment (aaronmarkham) [inline docs/tutorials/scala/mxnet_scala_on_intellij.md]: Can we get this added to the scala package so the build and docs build runs with less errors?

File: docs/tutorials/scala/mxnet_scala_on_intellij.md [hunk 3]
```diff
@@ -347,6 +379,30 @@ Library not loaded: /usr/local/opt/opencv/lib/libopencv_calib3d.x.x.dylib
...
 
 
+### Using MXNet from source
+
+If you chose to "Build from Source" when following the [install instructions](https://mxnet.incubator.apache.org/install/index.html), you can use your custom build instead of the build from maven.  Use your build by editing the `pom.xml` file and replacing the `org.apache.mxnet` dependency with the following:
+
+```
+      <groupId>org.apache.mxnet</groupId>
+      <artifactId>mxnet-core_${scala.version}-${platform}-sources</artifactId>
+      <version>1.3.0</version>
```
Comment (aaronmarkham) [inline docs/tutorials/scala/mxnet_scala_on_intellij.md]: maybe skip this?

File: docs/tutorials/scala/mxnet_scala_on_intellij.md [hunk 4]
```diff
@@ -347,6 +379,30 @@ Library not loaded: /usr/local/opt/opencv/lib/libopencv_calib3d.x.x.dylib
...
+      <scope>system</scope>
+      <systemPath>/PathToMXNetSource/incubator-mxnet/scala-package/assembly/osx-x86_64-cpu/target/mxnet-full_2.11-osx-x86_64-cpu-1.3.1-SNAPSHOT-sources.jar</systemPath>
+    </dependency>
+
+    <dependency>
+      <groupId>org.apache.mxnet</groupId>
+      <artifactId>mxnet-full_${scala.version}-${platform}</artifactId>
+      <version>1.3.0</version>
```
Comment (aaronmarkham) [inline docs/tutorials/scala/mxnet_scala_on_intellij.md]: skip version?

File: docs/tutorials/scala/mxnet_scala_on_intellij.md [hunk 5]
```diff
@@ -347,6 +379,30 @@ Library not loaded: /usr/local/opt/opencv/lib/libopencv_calib3d.x.x.dylib
 This can be resolved be installing OpenCV.
 
 
+### Using MXNet from source
+
+If you chose to "Build from Source" when following the [install instructions](https://mxnet.incubator.apache.org/install/index.html), you can use your custom build instead of the build from maven.  Use your build by editing the `pom.xml` file and replacing the `org.apache.mxnet` dependency with the following:
```
Comment (aaronmarkham) [inline docs/tutorials/scala/mxnet_scala_on_intellij.md]: There's this resource too: https://github.com/apache/incubator-mxnet/blob/master/docs/install/build_from_source.md

Comment (aaronmarkham) [reply docs/tutorials/scala/mxnet_scala_on_intellij.md]: And this table for bindings by OS. https://github.com/apache/incubator-mxnet/blob/master/docs/install/build_from_source.md#installing-mxnet-language-bindings

File: docs/tutorials/scala/mxnet_scala_on_intellij.md (hunk 3 above)
Comment (lanking520) [reply docs/tutorials/scala/mxnet_scala_on_intellij.md]: Yeah, don't have to include this here...

**Comment by Roshrini** [comment]
Thanks @zachgk for fixing the instructions. Can you please address the comments provided?

**Comment by Roshrini** [comment]
@nswamy Can you please add label [pr-awaiting-response]?

File: docs/tutorials/scala/mxnet_scala_on_intellij.md (hunk 2 above)
Comment (zachgk) [reply docs/tutorials/scala/mxnet_scala_on_intellij.md]: There are log4j.properties files already in the scala package. If you want to change how much they output, that should probably be separate from this and may require some discussion.

File: docs/tutorials/scala/mxnet_scala_on_intellij.md [hunk 6]
```diff
@@ -73,7 +73,6 @@ The configuration you should update is in the pom file's dependency for MXNet:
 <dependency>
   <groupId>org.apache.mxnet</groupId>
   <artifactId>mxnet-full_2.11-osx-x86_64-cpu</artifactId>
-  <version>1.2.0</version>
```
Comment (lanking520) [inline docs/tutorials/scala/mxnet_scala_on_intellij.md]: In fact, if you remove this, maven is not gonna work. As I tried last time, you still have to specify a version. You can simply change this as a placeholder and let the user find the right version number from Maven central

File: docs/tutorials/scala/mxnet_scala_on_intellij.md [hunk 7]
```diff
@@ -204,19 +203,25 @@ The project's `pom.xml` will be open for editing.
...
     <scala.compat.version>2.11</scala.compat.version>
+    <slf4jVersion>1.7.7</slf4jVersion>
+    <platform>osx-x86_64-cpu</platform>
   </properties>
 
   <dependencies>
     <dependency>
       <groupId>org.apache.mxnet</groupId>
       <artifactId>mxnet-full_2.11-osx-x86_64-cpu</artifactId>
-      <version>1.2.0</version>
```
Comment (lanking520) [inline docs/tutorials/scala/mxnet_scala_on_intellij.md]: Same applies here

**Pull request merged**